app/components/EmotionButton/EmotionButton.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { FlatList, Image, TouchableOpacity } from 'react-native';
+import _ from 'lodash';
+
+import { ExtendedText } from '../ExtendedText';
+import { EmotionButtonType } from './EmotionButton.data';
+import { IEmotionButton, IEmotionButtonProps } from './EmotionButton.types';
+import { styles } from './EmotionButton.styles';
+import { Nullable } from '../../utils';
+import { useAppSelector } from '../../hooks';
+
+export const EmotionButton: React.FC<IEmotionButtonProps> = ({
+  data,
+  setSelected,
+}) => {
+  const completedEmotions = useAppSelector(
+    state => state.cache.emotions.completed,
+  );
+  const [selectedItem, setSelectedItem] =
+    useState<Nullable<EmotionButtonType>>(null);
+  const [emotions, setEmotions] = useState<IEmotionButton[]>(data);
+
+  useEffect(() => {
+    setSelected(selectedItem);
+  }, [selectedItem, setSelected]);
+
+  useEffect(() => {
+    setEmotions(
+      _.compact(
+        _.map(data, item => {
+          if (_.indexOf(completedEmotions, item.type) === -1) {
+            return item;
+          }
+        }),
+      ),
```

Comment:
Can we use `_.flow` or `_.chain`?

Reply:
And also save `_,map` to variable and then work with it. Do not nest it inside itself many times